src/client/main.ts

```diff
@@ -266,3 +287,29 @@ export function addContextListener(handler: (context: Context) => void): Context
     contextListeners.push(listener);
     return listener;
 }
+
+/**
+ * Event that is fired whenever a window changes from one channel to another.
+ *
+ * This includes switching to/from the global channel. The `channel` and
+ * `previousChannel` fields use the same conventions for denoting the global channel as `getChannel`.
+ */
+export function addEventListener(event: 'channel-changed', handler: (event: ChannelChangedPayload) => void, identity?: Identity): EventListener;
+
+export function addEventListener(event: Event, handler: (event: EventPayload) => void, identity?: Identity): EventListener {
+    const listener = {event, handler, unsubscribe: () => {
+        const index: number = eventListeners.indexOf(listener);
+
+        if (index >= 0) {
+            eventListeners.splice(index, 1);
+        }
+
+        eventEmitter.removeListener(event, handler);
+        return index >= 0;
+    }};
+
+    eventListeners.push(listener);
+    eventEmitter.addListener(event, handler);
```

Comment:
How come we have two lists of event listeners? Can we just use `eventEmitter`, and drop `eventListeners`?

Assuming the latter is to help with testing? Not too familiar with that yet - lets follow-up.

Reply:
"Assuming the latter is to help with testing?" - yep, but I think it's changing anyway. Like, there's too competing standards to follow - stay in line with how addContextListener/addIntentListener work, or stay in line with how addEventListener works in layouts, and there's no nice way to satisfy both